Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -9,15 +9,21 @@ struct ConfirmPaymentSourceRequest: APIRequest {
     private let orderID: String
     private let pathFormat: String = "/v2/checkout/orders/%@/confirm-payment-source"
     private let accessToken: String
-    private let jsonEncoder = JSONEncoder()
+    var jsonEncoder: JSONEncoder?
     
     /// Creates a request to attach a payment source to a specific order.
     /// In order to use this initializer, the `paymentSource` parameter has to
     /// contain the entire dictionary as it exists underneath the `payment_source` key.
     init(
         accessToken: String,
-        cardRequest: CardRequest
+        cardRequest: CardRequest,
+        encoder: JSONEncoder? = nil
     ) throws {
+        if encoder != nil {
+            self.jsonEncoder = encoder
+        } else {
+            self.jsonEncoder = JSONEncoder()
+        }
```

Comment:
I think we could do the following here instead:
```suggestion
        self.jsonEncoder = encoder ?? JSONEncoder()
```

Reply:
addressed in [commit](4de094c2c79d020a4e248ad4c5677b8d55c02d2e)